src/Core/src/Application.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Maui
+{
+	public abstract class Application : IApplication
+	{
+		static object? GlobalLock;
+
+		IServiceProvider? _serviceProvider;
+		IMauiContext? _context;
+
+		protected Application()
+		{
+			GlobalLock = new object();
+
+			if (Current != null)
+				throw new InvalidOperationException($"Only one {nameof(Application)} instance is allowed");
+
+			lock (GlobalLock)
+			{
+				Current = this;
+			}
+		}
+
+		static public Application? Current { get; internal set; }
```

Comment:
We will require a `WindowCollection Windows` property and a `Window MainWindow` property. Not inlcuded in this PR, it will be added when adding MultiWindow support.

Reply:
I don't think we should have a `MainWindow` I agree in a WindowCollection for all the created Windows.

Reply:
谢谢大家的支持